iceoryx_hoofs/source/error_handling/error_handling.cpp

```diff
@@ -17,43 +17,24 @@
 
 #include "iceoryx_hoofs/error_handling/error_handling.hpp"
 
-#include "iceoryx_hoofs/log/logger.hpp"
-#include "iceoryx_hoofs/log/logging.hpp"
-#include "iceoryx_hoofs/log/logmanager.hpp"
-
-#include <iostream>
-#include <sstream>
-
 namespace iox
 {
-const char* ErrorHandler::ERROR_NAMES[] = {ICEORYX_ERRORS(CREATE_ICEORYX_ERROR_STRING)};
+const char* ERROR_NAMES[] = {ICEORYX_ERRORS(CREATE_ICEORYX_ERROR_STRING)};
 
 // NOLINTNEXTLINE(cert-err58-cpp) ErrorHander only used in tests
-HandlerFunction ErrorHandler::handler = {ErrorHandler::defaultHandler};
+iox::HandlerFunction ErrorHandler::handler = {ErrorHandler::defaultHandler};
 
-std::mutex ErrorHandler::handler_mutex;
-
-std::ostream& operator<<(std::ostream& stream, Error value) noexcept
+const char* toString(const Error error) noexcept
 {
-    stream << ErrorHandler::toString(value);
-    return stream;
+    return ERROR_NAMES[static_cast<typename std::underlying_type<Error>::type>(error)];
 }
 
-void ErrorHandler::defaultHandler(const Error error,
-                                  const std::function<void()>& errorCallBack,
-                                  const ErrorLevel level) noexcept
+void ErrorHandler::defaultHandler(const uint32_t, const char* errorName, const ErrorLevel level) noexcept
```

Comment:
Ok, the default handler does not care about the first parameter (I think it is basically the error code). Is this because it is already translated to `errorName`. The API may be a bit convoluted, but if so it already was before.

Reply:
True, this can probably optimised. Same for:

```
void errorHandler(const Error error,
                  const std::function<void()>& errorCallBack = std::function<void()>(),
                  const ErrorLevel level = ErrorLevel::FATAL) noexcept;
```

The 2nd and 3rd parameter will be swapped. Not on this PR.